cmd/create_device.go

```diff
@@ -60,58 +62,46 @@ packet device create --hostname [hostname] --plan [plan] --facility [facility_co
 
 `,
 	RunE: func(cmd *cobra.Command, args []string) error {
+		var endDt *packngo.Timestamp
 
-		request := &packngo.DeviceCreateRequest{
-			Hostname:     hostname,
-			Plan:         plan,
-			Facility:     []string{facility},
-			OS:           operatingSystem,
-			BillingCycle: billingCycle,
-			ProjectID:    projectID,
-		}
-
-		if storage != "" {
-			request.Storage = storage
-		}
-		if userdata != "" {
-			request.UserData = userdata
-		}
-
-		if customdata != "" {
-			request.CustomData = customdata
+		if userdata != "" && userdataFile != "" {
+			return fmt.Errorf("Either userdata or userdata-file should be set")
 		}
 
-		if len(tags) > 0 {
-			request.Tags = tags
-		}
-		if ipxescripturl != "" {
-			request.IPXEScriptURL = ipxescripturl
-		}
-		if publicIPv4SubnetSize != 0 {
-			request.PublicIPv4SubnetSize = publicIPv4SubnetSize
-		}
-		if alwaysPXE {
-			request.AlwaysPXE = alwaysPXE
-		}
-
-		if hardwareReservationID != "" {
-			request.HardwareReservationID = hardwareReservationID
-		}
-
-		if spotInstance {
-			request.SpotInstance = spotInstance
-		}
-
-		if spotPriceMax != 0 {
-			request.SpotPriceMax = spotPriceMax
+		if userdataFile != "" {
+			userdataRaw, readErr := ioutil.ReadFile(userdataFile)
+			if readErr != nil {
+				return errors.Wrap(readErr, "Could not read userdata-file")
+			}
+			userdata = string(userdataRaw)
```

Comment:
Might be good to add a check to see if the user specified both --userdata-file and --userdata to avoid any surprises if they specify both and only userdata-file is applied

Reply:
https://github.com/packethost/packet-cli/pull/88/files#diff-4dea4c1acc6ef47a6f4720eaa071fe90R67-R69

Reply:
:grimacing: apologies, multitasking got the better of me.